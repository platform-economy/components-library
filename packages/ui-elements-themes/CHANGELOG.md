# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
