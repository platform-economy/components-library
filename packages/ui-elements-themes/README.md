# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
