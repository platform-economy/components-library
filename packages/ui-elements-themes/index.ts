export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
