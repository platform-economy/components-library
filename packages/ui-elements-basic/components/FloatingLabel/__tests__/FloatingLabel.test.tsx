import * as React from 'react';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as TestRenderer from 'react-test-renderer';
import { withTestTheme } from '@relayr/ui-elements-themes';

import 'jest-styled-components';

enzyme.configure({ adapter: new Adapter() });
/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/

// Component
import { FloatingLabel } from '../index';

describe('ui-elements-basic', () => {
  describe('Components', () => {
    describe('FloatingLabel', () => {

      it('should render with default state', () => {
        const tree = TestRenderer
          .create(withTestTheme((
          <FloatingLabel>
            <input/>
            <label/>
          </FloatingLabel>
            )))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should render with props', () => {
        const tree = TestRenderer
          .create(withTestTheme((
          <FloatingLabel height={70} focused={true}>
            <input/>
            <label/>
          </FloatingLabel>
            )))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
