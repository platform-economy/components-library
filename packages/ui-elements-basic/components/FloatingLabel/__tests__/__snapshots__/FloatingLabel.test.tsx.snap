// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-basic Components FloatingLabel should render with default state 1`] = `
.c0 {
  position: relative;
}

.c0 input::-webkit-input-placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input::-moz-placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input:-ms-input-placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input::placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input:-ms-input-placeholder {
  opacity: 0;
}

.c0 label {
  position: absolute;
  will-change: transform;
  font-size: 1em;
  color: #text_inverted;
  pointer-events: none;
  -webkit-transition: all 0.3s cubic-bezier(.06,.67,.32,.82);
  transition: all 0.3s cubic-bezier(.06,.67,.32,.82);
  -webkit-transform: translate(10px,-25px);
  -ms-transform: translate(10px,-25px);
  transform: translate(10px,-25px);
}

.c0 div {
  margin-top: 15px;
}

.c0:focus-within label {
  font-size: .75em;
  color: #text;
}

.c0 input:not(:placeholder-shown) ~ label {
  font-size: .75em;
  color: #text;
}

@media screen and (min-width:414px) {
  .c0 label {
    font-size: .875em;
  }
}

<div
  className="c0"
>
  <input />
  <label />
</div>
`;

exports[`ui-elements-basic Components FloatingLabel should render with props 1`] = `
.c0 {
  position: relative;
}

.c0 input::-webkit-input-placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input::-moz-placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input:-ms-input-placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input::placeholder,
.c0 input::-webkit-input-placeholder {
  opacity: 0;
}

.c0 input:-ms-input-placeholder {
  opacity: 0;
}

.c0 label {
  position: absolute;
  will-change: transform;
  font-size: 1em;
  color: #text_inverted;
  pointer-events: none;
  -webkit-transition: all 0.3s cubic-bezier(.06,.67,.32,.82);
  transition: all 0.3s cubic-bezier(.06,.67,.32,.82);
  -webkit-transform: translate(10px,-25px);
  -ms-transform: translate(10px,-25px);
  transform: translate(10px,-25px);
  -webkit-transform: translate(0px,-55px);
  -ms-transform: translate(0px,-55px);
  transform: translate(0px,-55px);
  font-size: .75em;
  color: #text;
}

.c0 div {
  margin-top: 15px;
}

.c0:focus-within label {
  -webkit-transform: translate(0px,-55px);
  -ms-transform: translate(0px,-55px);
  transform: translate(0px,-55px);
  font-size: .75em;
  color: #text;
}

.c0 input:not(:placeholder-shown) ~ label {
  -webkit-transform: translate(0px,-55px);
  -ms-transform: translate(0px,-55px);
  transform: translate(0px,-55px);
  font-size: .75em;
  color: #text;
}

@media screen and (min-width:414px) {
  .c0 label {
    font-size: .875em;
  }
}

<div
  className="c0"
  height={70}
>
  <input />
  <label />
</div>
`;
